### PR TITLE
Feature/bael 4504 find available port

### PR DESCRIPTION
**Pull request by dstr89**
https://inprogress.baeldung.com/wp-admin/post.php?post=231187&action=edit&classic-editor

### CODE REVIEW

File: core-java-modules/core-java-networking-2/src/test/java/com/baeldung/socket/FindAvailablePortUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            assertThat(e).hasMessageContaining("Address already in use: bind");
+        }
+    }
+
+    @Test
+    public void givenExplicitPortRange_whenCreatingServerSocket_thenOnePortIsAssigned() {
+        for (int port : FREE_PORT_RANGE) {
+            try (ServerSocket serverSocket = new ServerSocket(port)) {
+                assertThat(serverSocket).isNotNull();
+                assertThat(serverSocket.getLocalPort()).isBetween(FREE_PORT_RANGE[0], FREE_PORT_RANGE[FREE_PORT_RANGE.length - 1]);
```
[the remark this answers is not in the document]
Comment (dstr89) [reply core-java-modules/core-java-networking-2/src/test/java/com/baeldung/socket/FindAvailablePortUnitTest.java]: Changed to isGreaterThan(0) as in other examples.

**Comment by lor6** [comment]
retest this please

**Comment by dstr89** [comment]
> retest this please

builds fine on my local, did a dummy commit now, because i cannot see jenkins build details

File: core-java-modules/core-java-networking-2/src/test/java/com/baeldung/socket/FindFreePortUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class FindFreePortUnitTest {
+
+    private static int FREE_PORT_NUMBER;
+    private static int[] FREE_PORT_RANGE;
+
+    @BeforeAll
+    public static void getExplicitFreePortNumberAndRange() {
+        try (ServerSocket serverSocket = new ServerSocket(0);) {
```
Comment (eric-martin) [inline core-java-modules/core-java-networking-2/src/test/java/com/baeldung/socket/FindFreePortUnitTest.java:24]: remove the semicolon at the end

File: core-java-modules/core-java-networking-2/src/test/java/com/baeldung/socket/FindFreePortUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            FREE_PORT_NUMBER = serverSocket.getLocalPort();
+            FREE_PORT_RANGE = new int[] {FREE_PORT_NUMBER, FREE_PORT_NUMBER + 1, FREE_PORT_NUMBER + 2};
+        } catch (IOException e) {
+            fail("No free port is available");
+        }
+    }
+
+    @Test
+    public void givenExplicitFreePort_whenCreatingServerSocket_thenThatPortIsAssigned() {
+        try (ServerSocket serverSocket = new ServerSocket(FREE_PORT_NUMBER);) {
```
Comment (eric-martin) [inline core-java-modules/core-java-networking-2/src/test/java/com/baeldung/socket/FindFreePortUnitTest.java:34]: remove the semicolon at the end

**Pull request closed without merging**